src/components/molecules/CreditingPeriodFilter/CreditingPeriodFilter.tsx

```diff
@@ -0,0 +1,135 @@
+  const { label, earliestDate, latestDate } = prop
+  const { setCreditingPeriodFilter } = useActions().searchFilters
+  const { selectedSearchFilterValues } = useAppState().searchFilters
+
+  const [minimumDate, setMinimumDate] = useState<Date | undefined>()
+  const [maximumDate, setMaximumDate] = useState<Date | undefined>()
+  const [showMinimumDateSelector, setShowMinimumDateSelector] = useState<boolean>(false)
+  const [showMaximumDateSelector, setShowMaximumDateSelector] = useState<boolean>(false)
+
+  const { t } = useTranslation(`search`)
+
+  useEffect(() => {
+    setMinimumDate(selectedSearchFilterValues.searchFilterValues.filterDates?.minDate)
+    setMaximumDate(selectedSearchFilterValues.searchFilterValues.filterDates?.maxDate)
+  }, [selectedSearchFilterValues.searchFilterValues.filterDates])
+
+  const renderDateSelector = (stateValue: Date | undefined, action: Dispatch<SetStateAction<Date | undefined>>) => {
+    return (
+      <CalendarWrapper
+        preSelectedDate={stateValue}
+        maxDate={latestDate}
+        minDate={earliestDate}
+        applySelectedDate={(selectedDate) => {
+          action(selectedDate)
+          setShowMinimumDateSelector(false)
+          setShowMaximumDateSelector(false)
+        }}
+      />
+    )
+  }
+
+  const renderContent = () => {
+    if (showMaximumDateSelector || showMinimumDateSelector) {
+      return null
+    }
+    return (
+      <VStack>
+        <HStack width={`100%`} justify={`center`} alignItems={`end`}>
+          <DateInput
+            label={`Minimum Date`}
+            value={minimumDate}
+            openDatePicker={() => setShowMinimumDateSelector(true)}
+            maxDate={latestDate}
+            minDate={earliestDate}
+            onChange={setMinimumDate}
+          />
+          <MinusIcon />
+          <DateInput
+            label={`Maximum Date`}
+            value={maximumDate}
+            openDatePicker={() => setShowMaximumDateSelector(true)}
+            maxDate={latestDate}
+            minDate={earliestDate}
+            onChange={setMaximumDate}
+          />
+        </HStack>
+        <HStack justify={`space-between`} width={`100%`} margin={`8px`}>
+          <Button
+            variant={`textLink`}
+            onClick={() => {
+              setMinimumDate(undefined)
+              setMaximumDate(undefined)
+              setCreditingPeriodFilter({})
+            }}
+            data-testid={`crediting-period-filter-clear`}
+          >
+            {t(`clear`)}
+          </Button>
+          <Button variant={`textLink`} onClick={() => setCreditingPeriodFilter({ maxDate: maximumDate, minDate: minimumDate })} data-testid={`crediting-period-filter-apply`}>
+            {t(`apply`)}
+          </Button>
+        </HStack>
+      </VStack>
+    )
+  }
+
+  const renderLabel = () => {
+    if (minimumDate && maximumDate) {
+      return `${format(minimumDate, `dd/MM/yyyy`)} - ${format(maximumDate, `dd/MM/yyyy`)}`
```

Comment:
Date patterns shouldn't be hard-corded. Please use the date patterns enum Mederic should have created.

Reply:
And please change that for all hard-coded date pattern in your files.